### ISSUE TITLE
How are "present" and "dismiss" methods are intended to be used with BottomSheetModals?

### ISSUE THREAD

**Issue by krakz999** [gorhom/react-native-bottom-sheet#336]
## Ask your Question

Hello,

So I'm using 2 modals on my screen, and I need to jump from one to the other and vice versa. After reading the documentation, at first I used the `dismiss()` method on the ref of the currently active modal, and `present()` on the next modal to switch, which worked, but once I dismissed the modal, I couldn't go back to it (same with `close()`), present doesn't work on those anymore.

The way it works for me is just calling `present()` on the modal I want to open, without dismissing. Is this the intended behaviour?

**My second issue that is kind of related:** When the user was on #1 snap point, went to the second modal, then went back  to the first with calling `present()`, the first modal would be on #0 snap point. I couldn't find a way to present the modal on #1 snap point instead of the default index. (please note in every other scenario my default has to be #0 snap point, but in this specific case I want it to open at #1. Is that currently implemented? 

I tried to use `snapTo(1)` after calling `present()` which only worked with a timeout, but it doesn't look good.

TLDR for the second: Can you call `present()` on a modal and animate it to a given snap index fluidly?

**Comment by gorhom** [comment]
>So I'm using 2 modals on my screen, and I need to jump from one to the other and vice versa. After reading the documentation, at first I used the dismiss() method on the ref of the currently active modal, and present() on the next modal to switch, which worked, but once I dismissed the modal, I couldn't go back to it (same with close()), present doesn't work on those anymore.

> The way it works for me is just calling present() on the modal I want to open, without dismissing. Is this the intended behaviour?

you do not need to dismiss the first modal, the library already handle that for you 

```
 present the first modal     ->     present the second modal     ->     dismiss the second modal
                                            |                                        |
                                   first modal will get minimise         first modal will be restored
```

> TLDR for the second: Can you call present() on a modal and animate it to a given snap index fluidly?

you can set `index` prop and when you call `present` it will animate to the provided index snap point

**Comment by krakz999** [comment]
> > So I'm using 2 modals on my screen, and I need to jump from one to the other and vice versa. After reading the documentation, at first I used the dismiss() method on the ref of the currently active modal, and present() on the next modal to switch, which worked, but once I dismissed the modal, I couldn't go back to it (same with close()), present doesn't work on those anymore.
> 
> > The way it works for me is just calling present() on the modal I want to open, without dismissing. Is this the intended behaviour?
> 
> you do not need to dismiss the first modal, the library already handle that for you
> 
> ```
>  present the first modal     ->     present the second modal     ->     dismiss the second modal
>                                             |                                        |
>                                    first modal will get minimise         first modal will be restored
> ```
> 
> > TLDR for the second: Can you call present() on a modal and animate it to a given snap index fluidly?
> 
> you can set `index` prop and when you call `present` it will animate to the provided index snap point

Thanks. Tried this already but for some reason `present` doesn't work on neither of the modals once I dismiss them  (or when the library dismisses them automatically).

**Comment by gorhom** [comment]
ahh ,,could you provide a reproducible sample code ? preferable from `/examples` folder

**Comment by krakz999** [comment]
So this is the issue I'm having with just using `present()` to let the library handle the modals automatically, and I'm able to reproduce it in the example project as well.

I have screen "A" and screen "B". I `present()` screen B from screen A, then I `present()` screen A from screen B to go back, after a while it stops working.

So what I'm trying to achieve is basically a list screen (screen A) and a details screen (screen B) but navigating between them breaks after a while.

Using the latest v2 of the package.

https://user-images.githubusercontent.com/22370549/110600194-f53c4a00-8183-11eb-92ae-b2d618efe8aa.mov

**Comment by krakz999** [comment]
I've pushed a fix for the stuck modal issue as part of this PR: https://github.com/gorhom/react-native-bottom-sheet/pull/337

https://user-images.githubusercontent.com/22370549/110621623-b960af00-819a-11eb-880e-9cf2d225d755.mov

**Comment by gorhom** [comment]
there is already pr , which refactor the modal handling https://github.com/gorhom/react-native-bottom-sheet/pull/327

**Comment by gorhom** [comment]
this should be fixed with `v2.2.4`, feel free to reopen this issue if it still not fixed
